File: sdkwork-ai-sdk-swift/Sources/API/RerankApi.swift
import Foundation

public class RerankApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Rerank documents
    public func rerank(body: RerankRequest) async throws -> RerankResponse? {
        let response = try await client.post(ApiPaths.aiPath("/rerank"), body: body)
        return response as? RerankResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/RerankApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class RerankApi
    {
        private readonly HttpClient _client;

        public RerankApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Rerank documents
        /// </summary>
        public async Task<RerankResponse?> RerankAsync(RerankRequest body)
        {
            return await _client.PostAsync<RerankResponse>(ApiPaths.AiPath("/rerank"), body);
        }
    }
}
